web_widget_many2one_kanban/static/src/js/web_widget_many2one_kanban.js

```diff
@@ -0,0 +1,58 @@
+/* Copyright 2019 Sunflower IT <www.sunflowerweb.nl>
+ * License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+ */
+
+odoo.define('web_widget_many2one_kanban.kanban_editable', function(require) {
+    "use strict";
+
+    var AbstractField = require('web.AbstractField');
+    var field_registry = require('web.field_registry');
+
+    var KanbanFieldMany2One = AbstractField.extend({
+        template: 'KanbanEditable',
+        supportedFieldTypes: ['many2one'],
+        events: _.extend({}, AbstractField.prototype.events, {
+            'click .oe_kanban_select': '_eventAction',
+            'change .oe_kanban_select': '_onSelectField',
+        }),
+        _render: function () {
+            var select = this.$('.oe_kanban_select');
+            var field_name = this.name;
+            var field_data = this.recordData[field_name];
+            var curr_id = field_data && field_data.data.id;
+            // Empty item
+            select.append('<option value=""></option>');
+            // Options
+            for (var i=0; i<this._options.length; i++) {
+                var selected = (this._options[i].id === curr_id)
+                    ? 'selected' : '';
+                var option = _.str.sprintf('<option value="%s" %s>%s</option>',
+                    this._options[i].id, selected, this._options[i].name);
+                select.append(option);
+            }
+            this._super.apply(this, arguments);
+            },
+        _onSelectField: function() {
+            var val = this.$('select[name="id"]').val();
+            val = this._parseValue(val).id;
+            this._setValue(val);
+            },
+        willStart: function() {
+            var self = this;
+            var model = this.field.relation;
+            return self._rpc({
+                model: model,
+                method: 'search_read',
+                fields: ['id', 'name'],
+                order: 'id',
+            }).then(function (data) {
+                self._options = data;
+            });
+            },
```

Comment:
We have a performance problem here: if 140 items are on the kanban board, this function will be executed 140 times, or 280 times if two such widgets are on one kanban. The loading drag on opening the kanban view is noticable. Possible solutions could involve:

1. On opening the kanban view, look for these kind of widgets in the individual kanbans, and if there are, load their options once and store them on the kanban view object. Then when displaying the kanbans, take the options from there; so we load just 2 times and not 280 times.
2. Study the Odoo source code and see if those options are not already loaded somewhere else and we can already pick them from the Javascript
3. On loading this widget first time, cache the field results somewhere on the main kanban view, and on subsequent widget loads, look at the cache first before querying RPC.